providers/windows/arch_windows.go

```diff
@@ -18,14 +18,46 @@
 package windows
 
 import (
-	windows "github.com/elastic/go-windows"
+	"golang.org/x/sys/windows"
+
+	gowindows "github.com/elastic/go-windows"
+)
+
+const (
+	imageFileMachineAmd64 = 0x8664
+	imageFileMachineArm64 = 0xAA64
+	archIntel             = "x86_64"
+	archArm64             = "arm64"
 )
 
 func Architecture() (string, error) {
-	systemInfo, err := windows.GetNativeSystemInfo()
+	systemInfo, err := gowindows.GetNativeSystemInfo()
 	if err != nil {
 		return "", err
 	}
 
 	return systemInfo.ProcessorArchitecture.String(), nil
 }
+
+func NativeArchitecture() (string, error) {
+	var processMachine, nativeMachine uint16
+	// the pseudo handle doesn't need to be closed
+	var currentProcessHandle = windows.CurrentProcess()
```

Comment:
Do we need to close this handle after we're done?

Reply:
no, as the explicit comment says. If you're not convinced check the [manual](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentprocess)

Reply:
Alright, we've had issues with leaving handles open, so I'm a tad paranoid...